ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp

```diff
@@ -267,15 +268,50 @@ bool window_get_stayontop() {
 }
 
 void window_set_sizeable(bool sizeable) {
-  if (enigma::isSizeable == sizeable) return;
   enigma::isSizeable = sizeable;
+  XSizeHints *sh = XAllocSizeHints();
+  sh->flags = PMinSize | PMaxSize;
+  
+  if (enigma::window_min_width == -1)
+    enigma::window_min_width = 1;
+  if (enigma::window_max_width == -1) 
+    enigma::window_max_width = display_get_width();
+  if (enigma::window_min_height == -1) 
+    enigma::window_min_height = 1;
+  if (enigma::window_max_height == -1) 
+    enigma::window_max_height = display_get_height();
+  
+  if (sizeable) {
+    sh->min_width = enigma::window_min_width;
+    sh->max_width = enigma::window_max_width;
+    sh->min_height = enigma::window_min_height;
+    sh->max_height = enigma::window_max_height;
+  } else {
+    sh->min_width = sh->max_width = window_get_width();
+    sh->min_height = sh->max_height = window_get_height();
+  }
+  XSetWMSizeHints(disp, win, sh, XA_WM_NORMAL_HINTS);
```

Comment:
After digging further about this myself, you could have actually just replaced this with XSetWMNormalHints which is what you need in the most cases. Consider doing that if it saves some lines.
https://stackoverflow.com/a/55837317

Reply:
i don't see how that is saving any lines. It's replacing 1 line with 1 line. I'm fine with doing it, I just don't see the difference if both reportedly do the same thing and the number of lines remains unaffected.

Reply:
The one I linked is actually what you already have. If you call XSetWMNormalHints, you don't need to construct/initialize that XAtom if I understand correctly. Look more closely.